src/pyggdrasil/interface/__init__.py

```diff
@@ -76,3 +80,71 @@ def append(self, iteration: int, tree: TreeNode, log_probability: float):
         self.iterations = jnp.append(self.iterations, iteration)
         self.trees.append(tree)
         self.log_probabilities = jnp.append(self.log_probabilities, log_probability)
+
+
+@dataclass
+class AugmentedMcmcData(PureMcmcData):
+    """Distance/Similarity and True Tree boolean enhanced MCMC data
+
+    intended to be human-readable and easy to use for plotting"""
+
+    similarity_measures: list[TreeSimilarityMeasure]
+    similarity_scores: list[jax.Array]
+    true_tree: list[bool]
+
+    def get_sample(
+        self, iteration: int
+    ) -> tuple[int, TreeNode, float, dict[TreeSimilarityMeasure, jax.Array], bool]:
+        """Return a sample from the MCMC chain.
+
+        Args:
+            iteration: iteration number
+        Returns:
+            tree: TreeNode object
+            log_probability: log-probability of the tree
+            similarity_scores: dictionary of similarity scores
+            true_tree: boolean indicating whether the tree is the true tree
+        """
+        return (
+            iteration,
+            self.trees[iteration],
+            self.log_probabilities[iteration].item(),
+            {
+                measure: self.similarity_scores[iteration][index].item()
+                for index, measure in enumerate(self.similarity_measures)
+            },
+            self.true_tree[iteration],
+        )
+
+    def append(
+        self,
+        iteration: int,
+        tree: TreeNode,
+        log_probability: float,
+        similarity_scores: Union[dict[TreeSimilarityMeasure, float]] = None,
+        true_tree: Optional[bool] = None,
```

Comment:
Am having a lot of trouble with overwriting / extending the function of the child class here. - Think I am missing something here. Think what I don't get is https://en.wikipedia.org/wiki/Liskov_substitution_principle - Pyright is giving me a hard time when I am assigning a `None` to a dictionary. What is a good way to make a dictionary optional ?